### PR TITLE
[#6785] categories/assets/category_formset: refactor category_formset

### PR DESCRIPTION
**Pull request by phillimorland**
Refactor to mostly remove jquery. It is still conditionally called for the select category icon element. After investigating alternatives, most options don't offer any advantage, the more accessible option that was found [here](https://w3c.github.io/aria-practices/examples/combobox/combobox-select-only.html) would require a custom django widget and is still not actually recommended due to possible gaps in support, we can look into it as a technical backlog story.

### CODE REVIEW

File: adhocracy4/categories/assets/category_formset.js [hunk 1]
```diff
@@ -1,82 +1,88 @@
...
+    let nextSibling = form.nextElementSibling
+    // console.log(form.children)
+    while (nextSibling && nextSibling !== this.formTemplate) {
+      const index = Array.from(this.formset.querySelectorAll('.js-form')).indexOf(nextSibling)
+      const indexTest = index - 1
+      console.log(index)
+      Array.from(nextSibling.firstElementChild.children).forEach(function (element, i) {
+        updateAttr(element, 'name', i)
+        updateAttr(element, 'for', i)
+        updateAttr(element, 'id', i)
```
Comment (phillimorland) [inline adhocracy4/categories/assets/category_formset.js:70]: `i` is actually the index of the field child elements, not the index of the parent element, so only the first element has it's attributes replaced as the following index's to not match `_old` variable from `updateAttr`

**Comment by phillimorland** [comment]
@goapunk should I tag in here?

**Comment by goapunk** [comment]
@phillimorland yes, please

**Review by goapunk — APPROVED** [review]
looks good to me!

**Pull request merged**